inc/commondbtm.class.php

```diff
@@ -5276,6 +5277,14 @@ function addFiles(array $input, $options = []) {
                'itemtype'      => $this->getType(),
                'items_id'      => $this->getID()
             ];
+
+            if ($date) {
+               // Set date, needed if it differs from the creation date
+               $toadd['date'] = $date;
+            } else {
+               $toadd['date'] = $_SESSION['glpi_currenttime'];
+            }
```

Comment:
If you apply Cedric's proposal:
```suggestion
            
            // Set date, needed if it differs from the creation date
            $toadd['date'] = $date ?? $_SESSION['glpi_currenttime'];
```